test/v1/controllers/requestParsers/validators/validations/MtdTaxYearValidationSpec.scala

```diff
@@ -16,39 +16,84 @@
 
 package v1.controllers.requestParsers.validators.validations
 
+import config.AppConfig
+import mocks.MockAppConfig
+import org.joda.time.DateTime
+import org.joda.time.format.{DateTimeFormat, DateTimeFormatter}
+import org.scalamock.handlers.CallHandler
 import support.UnitSpec
-import v1.models.errors.RuleTaxYearNotSupportedError
+import utils.CurrentDateTime
+import v1.mocks.MockCurrentDateTime
+import v1.models.errors.{RuleTaxYearNotEndedError, RuleTaxYearNotSupportedError}
 import v1.models.utils.JsonErrorValidators
 
 class MtdTaxYearValidationSpec extends UnitSpec with JsonErrorValidators {
 
+  class Test extends MockCurrentDateTime with MockAppConfig {
+    implicit val dateTimeProvider: CurrentDateTime = mockCurrentDateTime
+    val dateTimeFormatter: DateTimeFormatter = DateTimeFormat.forPattern("yyyy-MM-dd")
+
+    implicit val appConfig: AppConfig = mockAppConfig
+
+    def setupTimeProvider(date: String): CallHandler[DateTime] =
+      MockCurrentDateTime.getCurrentDate
+        .returns(DateTime.parse(date, dateTimeFormatter))
+
+    MockedAppConfig.minimumPermittedTaxYear
+      .returns(2021)
+  }
+
   "validate" should {
     "return no errors" when {
-      "a tax year greater than 2017 is supplied" in {
+      "the minimum allowed tax year is supplied" in new Test {
 
-        val validTaxYear = "2018-19"
-        val validationResult = MtdTaxYearValidation.validate(validTaxYear, RuleTaxYearNotSupportedError)
-        validationResult.isEmpty shouldBe true
+        setupTimeProvider("2022-04-06")
 
+        val validTaxYear = "2021-22"
+        val validationResult = MtdTaxYearValidation.validate(validTaxYear)
+        validationResult.isEmpty shouldBe true
       }
+      "the minimum allowed tax year is supplied with checkCurrentYear set to true" in new Test {
+
+        setupTimeProvider("2022-04-06")
 
-      "the minimum allowed tax year is supplied" in {
-        val validTaxYear = "2017-18"
-        val validationResult = MtdTaxYearValidation.validate(validTaxYear, RuleTaxYearNotSupportedError)
+        val validTaxYear = "2021-22"
```

Comment:
make the tests show that it runs now, use 2019-20 here and mock the dates currently used in app confing not future dates